sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala

```diff
@@ -21,360 +20,440 @@ import com.daml.lf.speedy.Compiler
+          ),
+          ValueRecord(
+            "Mod:Record",
+            ImmArray("fieldB" -> someText, "fieldA" -> aInt),
+          ),
+          ValueRecord(
+            "",
+            ImmArray("fieldA" -> aInt, "fieldB" -> someText),
+          ),
+          ValueRecord(
+            "Mod:Record",
+            ImmArray("" -> aInt, "fieldB" -> someText),
+          ),
+          ValueRecord(
+            "",
+            ImmArray("fieldB" -> someText, "fieldA" -> aInt),
+          ),
+        ),
+        SRecord(
+          "Mod:Record",
+          ImmArray("fieldA", "fieldB"),
+          ArrayList(SInt64(42), SOptional(Some(SText("42")))),
+        ),
       ),
       (
-        TTextMap(TBool),
-        ValueTextMap(SortedLookupList(Map("0" -> ValueTrue, "1" -> ValueFalse))),
-        SMap(true, Iterator(SText("0") -> SValue.True, SText("1") -> SValue.False)),
+        t"Mod:Variant Int64 Text",
+        ValueVariant("", "ConsB", ValueText("some test")),
+        List(
+          ValueVariant("Mod:Variant", "ConsB", ValueText("some test"))
+        ),
+        SVariant("Mod:Variant", "ConsB", 1, SText("some test")),
       ),
       (
-        TGenMap(TInt64, TText),
-        ValueGenMap(ImmArray(ValueInt64(1) -> ValueText("1"), ValueInt64(42) -> ValueText("42"))),
-        SMap(false, Iterator(SInt64(1) -> SText("1"), SInt64(42) -> SText("42"))),
+        Ast.TTyCon("Mod:Enum"),
+        ValueEnum("", "Cons1"),
+        List(
+          ValueEnum("Mod:Enum", "Cons1")
+        ),
+        SEnum("Mod:Enum", "Cons1", 0),
       ),
-      (TOptional(TText), ValueOptional(Some(ValueText("text"))), SOptional(Some(SText("text")))),
+    )
+
+    (pkg, testCases)
+  }
+
+  val upgradablePkgId = Ref.PackageId.assertFromString("-upgradable-v1-")
+
+  val (upgradablePkg, upgradableUserTestCases) = {
+    implicit val parserParameters: ParserParameters[ValueTranslatorSpec.this.type] =
+      ParserParameters(upgradablePkgId, LanguageVersion.v1_17)
+    val pkg = p"""metadata ( 'upgradable' : '1.0.0' )
+
+        module Mod {
+          record @serializable Record (a: *) (b: *) (c: *) (d: *) = { fieldA: a, fieldB: Option b, fieldC: Option c };
+          variant @serializable Variant (a: *) (b: *) (c: *) = ConsA: a | ConsB: b ;
+          enum @serializable Enum = Cons1 | Cons2 ;
+
+          record MyCons = { head : Int64, tail: Mod:MyList };
+          variant MyList = MyNil : Unit | MyCons: Mod:MyCons ;
+      }
+    """
+
+    val testCases = Table[Ast.Type, Value, List[Value], speedy.SValue](
+      ("type", "normalized value", "non normalized value", "svalue"),
       (
-        t"Mod:Tuple Int64 Text",
-        ValueRecord("", ImmArray("x" -> ValueInt64(33), "y" -> ValueText("a"))),
-        SRecord("Mod:Tuple", ImmArray("x", "y"), ArrayList(SInt64(33), SText("a"))),
+        t"Mod:Record Int64 Text Party Boolean Unit",
+        ValueRecord("", ImmArray("" -> aInt, "" -> none, "" -> someParty)),
+        List(
```

Comment:
As discussed over zoom, add a case where the package ID doesn't resolve to the same package name and show that it is ignored as intended. Same below for variants and enums.